src/components/autocomplete/js/autocompleteController.js

```diff
@@ -466,9 +466,9 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        event.preventDefault(); // Prevent browser from always clearing input
```

Comment:
`preventDefault` stops the default browser event
`stopPropagation` stops the event from bubbling up the DOM tree

Some browsers (ie: Chrome), on `input` elements with `type` set to `search` will clear the `value`. Since we're handling clearing selectively, we're not letting that event pass to the browser.

We are letting ESC to propagate up the DOM tree so other elements can handle it. For example, we want dialogs to close on ESC keydown.